src/databricks/labs/ucx/assessment/export.py

```diff
@@ -0,0 +1,116 @@
+
+
+class Exporter:
+    # File and Path Constants
+    _ZIP_FILE_NAME = "ucx_assessment_results.zip"
+
+    def __init__(self, ctx: WorkspaceContext):
+        self._ctx = ctx
+
+    def _get_ucx_main_queries(self) -> list[dict[str, str]]:
+        """Retrieve and construct the main UCX queries."""
+        pattern = r"\b.inventory\b"
+        schema = self._ctx.inventory_database
+        project_root = Path(__file__).parent.parent.parent.parent
+        ucx_main_queries_path = project_root / "labs/ucx/queries/assessment/main"
+
+        # List all SQL files in the directory, excluding those with 'count' in their names
+        sql_files = [file for file in ucx_main_queries_path.iterdir() if file.suffix == ".sql"]
+
+        ucx_main_queries = []
+
+        for sql_file in sql_files:
+            content = sql_file.read_text()
+            modified_content = re.sub(pattern, f" {schema}", content, flags=re.IGNORECASE)
+            query_name = sql_file.stem
+            ucx_main_queries.append({"name": query_name, "query": modified_content})
+
+        return ucx_main_queries
+
+    @staticmethod
+    def _extract_target_name(name: str, pattern: str) -> str:
+        """Extract target name from the file name using the provided pattern."""
+        match = re.search(pattern, name)
+        return match.group(1) if match else ""
+
+    @staticmethod
+    def _cleanup(path: Path, target_name: str) -> None:
+        """Remove a specific CSV file in the given path that matches the target name."""
+        target_file = path.joinpath(target_name)
+
+        if target_file.exists():
+            target_file.unlink()
+
+    def _execute_query(self, path: Path, result: dict[str, str]) -> None:
+        """Execute a SQL query and write the result to a CSV file."""
+        pattern = r"^\d+_\d+_(.*)"
+        match = re.search(pattern, result["name"])
+        if match:
+            file_name = f"{match.group(1)}.csv"
+            csv_path = os.path.join(path, file_name)
+
+            query_results = list(self._ctx.sql_backend.fetch(result["query"]))
+
+            if query_results:
+                headers = query_results[0].asDict().keys()
+                with open(csv_path, mode='w', newline='', encoding='utf-8') as file:
+                    writer = csv.DictWriter(file, fieldnames=headers)
+                    writer.writeheader()
+                    for row in query_results:
+                        writer.writerow(row.asDict())
+                # Add the CSV file to the ZIP archive
+                self._add_to_zip(path, file_name)
+
+    def _add_to_zip(self, path: Path, file_name) -> None:
+        """Create a ZIP file containing all the CSV files."""
+        zip_path = path / self._ZIP_FILE_NAME
+        file_path = path / file_name
+
+        try:
+            with ZipFile(zip_path, 'a') as zipf:
+                zipf.write(file_path, arcname=file_name)
+
+        except FileNotFoundError:
+            print(f"File {file_path} not found.")
```

Comment:
```suggestion
            logger.warning(f"File {file_path} not found.")
```
don't use `print`, use a logger; i've realised you didn't run `make fmt`, as our static code analysis checker would catch it on your local machine instead.